### PR TITLE
Logging and Longrun introduced in Upgrade Automation

### PR DESCRIPTION
**Pull request by jyejare**
This PR contains:
1. The massive Logging facility and change that help makes Upgrade Job users job easy. This will create two log files:
      1. upgrade_highlights : The special new highlight logging level logs will be logged to this file which will tell us the Upgrade status in one look.
     2. full_upgrade: All the logs of all levels will be logged to this file for debugging purpose.
2. Longrun support to Upgrade Satellite, Capsule and clients at a time.
3. New Classes and methods for Logging
4. Some existing functions are modified/improved for better logging
5. Some minor fixes not related to logging

### CODE REVIEW

**Comment by jyejare** [comment]
This idea has been triggered by this Robotello-ci issue: https://github.com/SatelliteQE/robottelo-ci/issues/388
Creating the same issue here for tracking purpose : https://github.com/SatelliteQE/automation-tools/issues/510

**Comment by jyejare** [comment]
closes #510

**Comment by jyejare** [comment]
As I reffered 2 robottelo-ci PR's above are highly dependent on this PR.

After reviewing and merging this PR, please review those PR's .

**Comment by jyejare** [comment]
@SatelliteQE/quality-engineers , Guys please visit this PR which has enhancement the logging of Upgrade Automation. Now your personal and Triggered systems Upgrade Status directly into your mailbox :)

**Comment by jyejare** [comment]
closes #499

File: automation_tools/satellite6/upgrade/capsule.py [hunk 1]
```diff
@@ -154,11 +166,12 @@ def satellite6_capsule_zstream_upgrade():
     TO_VERSION
         Next satellite version to which satellite will be upgraded
     """
+    logger.highlight('\n========== CAPSULE UPGRADE =================\n')
     from_version = os.environ.get('FROM_VERSION')
     to_version = os.environ.get('TO_VERSION')
     if not from_version == to_version:
-        print ('Error! zStream Upgrade on Capsule cannot be performed as '
-               'FROM and TO versions are not same!')
+        logger.warning('zStream Upgrade on Capsule cannot be performed as '
```
Comment (sghai) [inline automation_tools/satellite6/upgrade/capsule.py]: shouldn't this be an error message ?

File: automation_tools/satellite6/upgrade/client.py [hunk 2]
```diff
@@ -9,13 +10,25 @@
...
     remove_all_docker_containers,
     sync_tools_repos_to_upgrade
 )
+from automation_tools.satellite6.upgrade.tools import logger
+from fabric.api import env, execute, run
+
+logger = logger()
+
+
+def ver_filter(rpm_name):
```
Comment (sghai) [inline automation_tools/satellite6/upgrade/client.py]: `get_agent_version` ? or `version_filter` ?

File: automation_tools/satellite6/upgrade/capsule.py (hunk 1 above)
Comment (jyejare) [inline automation_tools/satellite6/upgrade/capsule.py]: I believe to throw an error message, the error has to be in Product and shouldnt be in automation or configuration.

File: automation_tools/satellite6/upgrade/satellite.py [hunk 3]
```diff
@@ -71,18 +75,19 @@ def satellite6_upgrade():
...
         e.g '6.1','6.2'
     """
+    logger.highlight('\n========== SATELLITE UPGRADE =================\n')
     to_version = os.environ.get('TO_VERSION')
     rhev_sat_host = os.environ.get('RHEV_SAT_HOST')
     base_url = os.environ.get('BASE_URL')
     if to_version not in ['6.1', '6.2']:
-        print('Wrong Satellite Version Provided to upgrade to. '
-              'Provide one of 6.1, 6.2')
+        logger.warning('Wrong Satellite Version Provided to upgrade to. '
```
Comment (sghai) [inline automation_tools/satellite6/upgrade/satellite.py]: shouldn't it be a error log ?

Comment (jyejare) [inline automation_tools/satellite6/upgrade/satellite.py]: Refer my above comment for same reason.

**Comment by sghai** [comment]
A few pending comments warning vs error logs otherwise code looks sane to me..

**Comment by sghai** [comment]
ACK..
@kbidarkar @lpramuk : Hey Guys, could anyone of you please review the PR ?

**Comment by jyejare** [comment]
@SatelliteQE/quality-engineers , Hey guys this one is pending from long. Please visit soon as you can. Its huge plus in Upgrade Automation Status.

File: automation_tools/satellite6/upgrade/tools.py [hunk 4]
```diff
@@ -3,11 +3,84 @@
...
 import time
+
 from fabric.api import execute, run
 
 
+HIGHLIGHT_LEVEL_NUM = 25
+logging.addLevelName(HIGHLIGHT_LEVEL_NUM, 'HIGHLIGHT')
+
+
+class MyLogger(logging.Logger):
```
Comment (rplevka) [inline automation_tools/satellite6/upgrade/tools.py]: your logger? :)

Comment (jyejare) [inline automation_tools/satellite6/upgrade/tools.py]: Nice catch :)

**Pull request merged**